File: backend/src/docfabric/models/document.py
from datetime import datetime
from enum import Enum
from uuid import UUID

from pydantic import BaseModel


class DocumentStatus(str, Enum):
    processing = "processing"
    ready = "ready"
    error = "error"


class DocumentMetadata(BaseModel):
    id: UUID
    filename: str
    content_type: str
    size_bytes: int
    status: DocumentStatus
    metadata: dict[str, str]
    error: str | None = None
    created_at: datetime
    updated_at: datetime


class DocumentList(BaseModel):
    items: list[DocumentMetadata]
    total: int
    limit: int
    offset: int


class DocumentContent(BaseModel):
    content: str
    total_length: int
    offset: int
    length: int


class OutlineMode(str, Enum):
    flat = "flat"
    nested = "nested"


class OutlineSection(BaseModel):
    level: int
    title: str
    heading_path: str
    offset: int
    length: int


class DocumentOutline(BaseModel):
    sections: list[OutlineSection]
    total_length: int

File: frontend/src/types/document.ts
export type DocumentStatus = "processing" | "ready" | "error";

export interface DocumentMetadata {
  id: string;
  filename: string;
  content_type: string;
  size_bytes: number;
  status: DocumentStatus;
  metadata: Record<string, string>;
  error: string | null;
  created_at: string;
  updated_at: string;
}

export interface DocumentList {
  items: DocumentMetadata[];
  total: number;
  limit: number;
  offset: number;
}

export interface DocumentContent {
  content: string;
  total_length: number;
  offset: number;
  length: number;
}
